### PR TITLE
dev/financial#184 Fix currency name for Ghana and Belarus

### PR DESCRIPTION
**Pull request by mlutfy**
Overview
----------------------------------------

https://lab.civicrm.org/dev/financial/-/issues/184

To reproduce:

* Go to dmaster
* Admin > Localisation > Language etc
* Set the default currency to "GHC"

Result:

> Brick\Money\Exception\UnknownCurrencyException: Unknown currency code: GHC in Brick\Money\Exception\UnknownCurrencyException::unknownCurrency() (line 19 of /srv/buildkit/build/dmaster/web/sites/all/modules/civicrm/vendor/brick/money/src/Exception/UnknownCurrencyException.php).

The ISO code for Ghana is "GHS", which is what Brick/Money uses.

https://en.wikipedia.org/wiki/Ghanaian_cedi

Technical Details
----------------------------------------

The problem surfaced when upgrading to the Brick/Money library. The old values were not technically wrong, in the year when CiviCRM was first created.

- Ghana adopted the Second cedi (GHC) in 1967 until 2007, and then it was replaced by the Third cedi (GHS) after "rampant inflation" (according to Wikipedia).
- Belarus has been using BYN since 2016, replacing the BYR.

The proper thing to do would have been to add a new currency and phase out the old one, but in this specific case, I think it's safe to say that most CiviCRM installations, if they are using those currencies, probably have already converted their accounting to the newer currencies.

I wrote a generic-ish function, knowing it might not be used very often, but I figure that my clients are pretty good at using currencies that not many people are using, so if we find any, I will re-use that function. In the situations I encountered, they did not have any financial transactions in the CRM.

### CODE REVIEW

File: CRM/Upgrade/Incremental/Base.php [hunk 1]
```diff
@@ -305,6 +305,39 @@ public static function updatePrintLabelToken($ctx, string $old, string $new, $ve
     return TRUE;
   }
 
+  /**
+   * Updated a currency in civicrm_currency and related configurations
+   *
+   * @param CRM_Queue_TaskContext $ctx
+   * @param string $old
+   * @param string $new
```
[the remark this answers is not in the document]
Comment (mlutfy) [reply CRM/Upgrade/Incremental/Base.php]: @totten Thanks!

**Comment by mlutfy** [comment]
rebased

**Comment by totten** [comment]
Alternatively, if it's something quite subjective - then maybe it's fair to take a best-guess at what transition will be needed most often, and add an upgrade-message to alert anyone who might have a funny case.

**Comment by mlutfy** [comment]
@yashodha Thanks! Should be fixed.

**Comment by colemanw** [comment]
@mlutfy actually now it's 5.45. If the feedback is all addressed then we can go ahead and merge this once it's rebased onto the latest version.

**Comment by yashodha** [comment]
@mlutfy looks good, can be merged after upgrade version fixed to 5.45.

**Comment by mlutfy** [comment]
Rebased on master, and moved the upgrade to FiveFortyFive.php

**Comment by eileenmcnaughton** [comment]
test fail is unrelated - failing elsewhere=, date based

**Pull request merged**